src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -317,6 +335,254 @@ def on_time_step__prepare(self, event: Event):
+        child_rp_idx = household_structure.loc[
+            (household_structure["relation_to_household_head_x"] == "Reference person")
+            & (household_structure["age_x"] < 18)
+        ].index
+
+        # Potential guardian/household member index groups
+        relatives_of_rp_idx = household_structure.loc[
+            ~household_structure["relation_to_household_head_y"].isin(non_relatives)
+        ].index
+        non_relatives_of_rp_idx = household_structure.loc[
+            household_structure["relation_to_household_head_y"].isin(non_relatives)
+        ].index
+        age_bound_idx = household_structure.loc[
+            (household_structure["age_difference"] >= 20)
+            & (household_structure["age_difference"] < 46)
+        ].index
+        parents_of_rp_idx = household_structure.loc[
+            household_structure["relation_to_household_head_y"].isin(parents)
+        ].index
+        over_18_idx = household_structure.loc[household_structure["age_y"] >= 18].index
+        rp_idx = household_structure.loc[
+            household_structure["relation_to_household_head_y"] == "Reference person"
+        ].index
+        partners_idx = household_structure.loc[
+            household_structure["relation_to_household_head_y"].isin(partners)
+        ].index
+
+        # Assign guardians across groups
+        # Children of reference person - red box
+        rp_parent_ids = (
+            household_structure.loc[children_of_rp_idx.intersection(rp_idx)]
+            .reset_index()
+            .set_index("child_id")["person_id"]
+        )
+        pop.loc[rp_parent_ids.index, "guardian_1"] = rp_parent_ids
+        # Assign partners of reference person as partners
+        partners_of_rp_parent_ids = household_structure.loc[
+            children_of_rp_idx.intersection(partners_idx)
+        ]
+        if len(partners_of_rp_parent_ids) > 0:
+            partners_of_rp_parent_ids = (
+                partners_of_rp_parent_ids.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[partners_of_rp_parent_ids.index, "guardian_2"] = partners_of_rp_parent_ids
+
+        # Children are relative of reference person with relative(s) in age bound - orange box
+        relatives_with_guardian = household_structure.loc[
+            child_relative_of_rp_idx.intersection(relatives_of_rp_idx).intersection(
+                age_bound_idx
+            )
+        ]
+        if len(relatives_with_guardian) > 0:
+            relatives_with_guardian_ids = (
+                relatives_with_guardian.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[
+                relatives_with_guardian_ids.index, "guardian_1"
+            ] = relatives_with_guardian_ids
+
+        # Children are relative of reference person with no age bound relative(s) - yellow box
```

Comment:
in addition to the "yellow box" comment here it would be helpful to include what the assignment is. like: 
`Children are relative of reference person with no age bound relative(s) - yellow box, assign reference person`

so that if youre going through this you dont have to also have the diagram up to know what this snippet of code is assigning

Reply:
--- fwiw the others were more readable to me, this one is hard to understand what you're doing